### ISSUE TITLE
Fix false negatives for nested selectors in selector-combinator-blacklist

### ISSUE THREAD

**Issue by avinashbot** [stylelint/stylelint#4582]
> Clearly describe the bug

I'm using the `selector-combinator-blacklist` rule to blacklist the " " (descendant) combinator. This works fine for rules like this:

```scss
a b { } // Unexpected combinator " "
```

But it misses the implicit descendant combinator that results from nesting without an explicit combinator:

```scss
// This is allowed...
a {
  b {
    ...
  }
}

// ...but it compiles into this, which isn't
a b { ... }
```

I did try using `selector-nested-pattern` rule to make this work, but trying to craft a regex to match all possible combination of selectors but one is a crazy ordeal.

For performance reasons, this might have to be its own rule, but at least in the case of nesting, this rule misses a very clear use-case. I wasn't sure if this is a bug report or a feature request, but I felt that since there's a good chance this would lead to unintended styling issues, I went with a bug report.

> Which rule, if any, is the bug related to?

`selector-combinator-blacklist` with `[" "]`, but it presumably also applies to `selector-combinator-whitelist` (if I whitelist everything except " ")

> What CSS is needed to reproduce the bug?

```scss
a {
  b {
    width: 00.10em;
  }
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "selector-combinator-blacklist": [" "],
  }
}
```

> Which version of stylelint are you using?

`12.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.scss" --config myconfig.json`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS nested properties.

> What did you expect to happen?

A warning to be flagged for blacklisted selector.

> What actually happened (e.g. what warnings or errors did you get)?

No warnings.

**Comment by jeddy3** [comment]
@avinashbot Thanks for the clear report and for using the template.

> I wasn't sure if this is a bug report or a feature request, but I felt that since there's a good chance this would lead to unintended styling issues, I went with a bug report.

I think you were right to go with a bug report.

The similar [`selector-no-qualifying-type` rule](https://github.com/stylelint/stylelint/tree/master/lib/rules/selector-no-qualifying-type) resolves nested selectors, and I believe the `selector-combinator-blacklist` rule should too.

I've labelled the issue as ready to implement. Please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) if you have time. You can use the [code](https://github.com/stylelint/stylelint/blob/master/lib/rules/selector-no-qualifying-type/index.js#L109-L116) and [tests](https://github.com/stylelint/stylelint/blob/master/lib/rules/selector-no-qualifying-type/__tests__/index.js#L2139-L2239) within `selector-no-qualifying-type` to guide you.

We're also improving our [docs to help contributors](https://github.com/stylelint/stylelint/blob/de6f8ebdb6c4e7c49d790dff7a607112ae12792a/CONTRIBUTING.md#code-contributions) (and this will land soon).

**Comment by jeddy3** [comment]
@fanich37 Thanks for the "syntax: scss" label. I had overlooked that this is only an issue for implicit nesting. This rule works as expected for standard selectors that explicitly use the _nesting selector_ (`&`):

```css
a {
  & b {
    color: #FFF;
  }
}
```

Produces:

```shell
 2:4 error Unexpected combinator " " (selector-combinator-blacklist)
```

SCSS variants of the `selector-combinator-white/blacklist` built-in rules, that support implicit nesting, belong in [stylelint-scss](https://github.com/kristerkari/stylelint-scss) - a plugin pack of rules dedicated to SCSS-like syntax.

@avinashbot You can either use explicit nesting or create and contribute a `scss/selector-combinator-blacklist` plugin to [stylelint-scss](https://github.com/kristerkari/stylelint-scss).
